edit-post/components/plugins-panel/index.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Panel, PanelBody, IconButton, withFocusReturn } from '@wordpress/components';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+import { getSidebarSettings } from '../../api/sidebar';
+import { getActivePlugin } from '../../store/selectors';
+import { closeGeneralSidebar } from '../../store/actions';
+
+/**
+ * Returns the sidebar that should be rendered in the sidebar registered by
+ * plugins.
+ *
+ * @param {string} plugin The currently active plugin.
+ *
+ * @return {Object} The React element to render as a panel.
+ */
+export function getPluginSidebar( plugin ) {
+	const pluginSidebar = getSidebarSettings( plugin );
+
+	if ( ! pluginSidebar ) {
+		return {
+			title: __( 'Error: Unregistered plugin requested.' ),
+			render: () => {
+				return <Panel>
+					<PanelBody>
+						{ sprintf( __( 'No matching plugin sidebar found for plugin "%s"' ), plugin ) }
+					</PanelBody>
+				</Panel>;
+			},
+		};
+	}
+	return pluginSidebar;
+}
+
+function PluginsPanel( { onClose, plugin } ) {
+	const {
+		title,
+		render,
+	} = getPluginSidebar( plugin );
+	return (
+		<div
```

Comment:
Based on my comment https://github.com/WordPress/gutenberg/pull/4777/files/3cdf86f520f8f379b50eeed62872e24fb7516b76#r167880008 below, this would be something for a next iteration as well (since <Header/> is nested inside <Sidebar/>)